src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -247,6 +248,19 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('days_retained', type=int,
                    help='Number of days that soft-deleted blob will be retained. Must be in range [1,365].')
 
+    with self.argument_context('storage blob service-properties update', min_api='2018-03-28') as c:
+        c.argument('delete_retention', arg_type=get_three_state_flag(), arg_group='Soft Delete',
+                   help='Enables soft-delete.')
+        c.argument('days_retained', type=int, arg_group='Soft Delete',
+                   help='Number of days that soft-deleted blob will be retained. Must be in range [1,365].')
```

Comment:
This should be something like `--retention` so that the units are not in the parameter name.

Reply:
That would be confusing with `--delete-retention`

Reply:
I would go with just `--days` like `advisor recommendation disable` and some monitor command I've seen except that service-properties look like:
```
{
  "cors": [],
  "deleteRetentionPolicy": {
    "days": null,
    "enabled": false
  },
  "hourMetrics": {
    "enabled": true,
    "includeApis": true,
    "retentionPolicy": {
      "days": 7,
      "enabled": true
    },
    "version": "1.0"
  },
  "logging": {
    "delete": false,
    "read": false,
    "retentionPolicy": {
      "days": null,
      "enabled": false
    },
    "version": "1.0",
    "write": false
  },
```